contracts/cosmwasm-vm/cw-xcall/src/execute_call.rs

```diff
@@ -81,18 +81,23 @@ impl<'a> CwCallService<'a> {
         let (response, event) = match msg.result {
             cosmwasm_std::SubMsgResult::Ok(_res) => {
                 let code = CallServiceResponseType::CallServiceResponseSuccess.into();
-
+                let reply = self
+                    .pop_call_reply(deps.storage, request.to().clone())
+                    .map(|msg| rlp::encode(&msg).to_vec());
                 let message_response = CSMessageResult::new(
                     request.sequence_no(),
                     CallServiceResponseType::CallServiceResponseSuccess,
+                    reply,
                 );
+
                 let event = event_call_executed(req_id, code, "success");
                 (message_response, event)
             }
             cosmwasm_std::SubMsgResult::Err(err) => {
                 let code = CallServiceResponseType::CallServiceResponseFailure;
                 let error_message = format!("CallService Reverted : {err}");
-                let message_response = CSMessageResult::new(request.sequence_no(), code.clone());
+                let message_response =
```

Comment:
Do we not need to cleanup/remove the reply here if there is one set?

Reply:
Or mabye that cant happend

Reply:
ya we are popping the reply from storage.